types/parser/parse.go

```diff
@@ -0,0 +1,111 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/antlr4-go/antlr/v4"
+	substraitgo "github.com/substrait-io/substrait-go"
+	"github.com/substrait-io/substrait-go/types"
+	baseparser2 "github.com/substrait-io/substrait-go/types/parser/baseparser"
+)
+
+type TypeExpression struct {
+	ValueType types.FuncDefArgType
+}
+
+func (t *TypeExpression) MarshalYAML() (interface{}, error) {
+	return t.ValueType.String(), nil
+}
+
+func (t *TypeExpression) UnmarshalYAML(fn func(interface{}) error) error {
+	type Alias any
+	var alias Alias
+	if err := fn(&alias); err != nil {
+		return err
+	}
+
+	switch v := alias.(type) {
+	case string:
+		exp, err := ParseType(v)
+		if err != nil {
+			return err
+		}
+		t.ValueType = exp
+		return nil
+	}
+
+	return substraitgo.ErrNotImplemented
+}
+
+type simpleErrorListener struct {
+	errorCount int
+	errors     []string
+}
+
+func (l *simpleErrorListener) SyntaxError(recognizer antlr.Recognizer, offendingSymbol interface{}, line, column int, msg string, e antlr.RecognitionException) {
+	l.errorCount++
+	l.errors = append(l.errors, fmt.Sprintf("Syntax error at line %d:%d: %s ", line, column, msg))
+}
+
+func (l *simpleErrorListener) ReportAmbiguity(recognizer antlr.Parser, dfa *antlr.DFA, startIndex, stopIndex int, exact bool, ambigAlts *antlr.BitSet, configs *antlr.ATNConfigSet) {
+}
+
+func (l *simpleErrorListener) ReportAttemptingFullContext(recognizer antlr.Parser, dfa *antlr.DFA, startIndex, stopIndex int, conflictingAlts *antlr.BitSet, configs *antlr.ATNConfigSet) {
+}
+
+func (l *simpleErrorListener) ReportContextSensitivity(recognizer antlr.Parser, dfa *antlr.DFA, startIndex, stopIndex, prediction int, configs *antlr.ATNConfigSet) {
+}
+
+func (l *simpleErrorListener) ErrorCount() int {
+	return l.errorCount
+}
+
+func (l *simpleErrorListener) GetError() string {
```

Comment:
This shouldn't be called if ErrorCount() is not positive.  I'd argue that the caller can easily get this on their own since they're also the one doing the count check.   Remove it?